tests/integration/s3/test_s3.py

```diff
@@ -245,175 +516,2240 @@ def test_put_object_checksum(self, s3_client, s3_create_bucket, algorithm):
+                    }
+                ]
+            },
+        )
+
+        response = s3_client.get_bucket_cors(Bucket=s3_bucket)
+        snapshot.match("get-bucket-cors", response)
+
+        result = s3_client.get_bucket_acl(Bucket=s3_bucket)
+        snapshot.match("get-bucket-acl", result)
+
+        with pytest.raises(ClientError) as e:
+            s3_client.get_bucket_acl(Bucket="bucket-not-exists")
+        snapshot.match("get-bucket-not-exists", e.value.response)
+
+    @pytest.mark.aws_validated
+    @pytest.mark.skip_snapshot_verify(
+        paths=["$..VersionId", "$..ContentLanguage", "$..Error.RequestID"]
+    )
+    def test_s3_uppercase_key_names(self, s3_client, s3_create_bucket, snapshot):
+        # bucket name should be case-sensitive
+        bucket_name = f"testuppercase-{short_uid()}"
+        s3_create_bucket(Bucket=bucket_name)
+
+        # key name should be case-sensitive
+        object_key = "camelCaseKey"
+        s3_client.put_object(Bucket=bucket_name, Key=object_key, Body="something")
+        res = s3_client.get_object(Bucket=bucket_name, Key=object_key)
+        snapshot.match("response", res)
+        with pytest.raises(ClientError) as e:
+            s3_client.get_object(Bucket=bucket_name, Key="camelcasekey")
+        snapshot.match("wrong-case-key", e.value.response)
+
+    @pytest.mark.aws_validated
+    def test_s3_download_object_with_lambda(
+        self,
+        s3_client,
+        s3_create_bucket,
+        create_lambda_function,
+        lambda_client,
+        lambda_su_role,
+        logs_client,
+    ):
+
+        bucket_name = f"bucket-{short_uid()}"
+        function_name = f"func-{short_uid()}"
+        key = f"key-{short_uid()}"
+
+        s3_create_bucket(Bucket=bucket_name)
+        s3_client.put_object(Bucket=bucket_name, Key=key, Body="something..")
+
+        create_lambda_function(
+            handler_file=os.path.join(
+                os.path.dirname(__file__),
+                "../awslambda",
+                "functions",
+                "lambda_triggered_by_sqs_download_s3_file.py",
+            ),
+            func_name=function_name,
+            role=lambda_su_role,
+            runtime=LAMBDA_RUNTIME_PYTHON39,
+            envvars=dict(
+                {
+                    "BUCKET_NAME": bucket_name,
+                    "OBJECT_NAME": key,
+                    "LOCAL_FILE_NAME": "/tmp/" + key,
+                }
+            ),
+        )
+        lambda_client.invoke(FunctionName=function_name, InvocationType="Event")
+
+        # TODO maybe this check can be improved (do not rely on logs)
+        retry(
+            check_expected_lambda_log_events_length,
+            retries=10,
+            sleep=1,
+            function_name=function_name,
+            regex_filter="success",
+            expected_length=1,
+            logs_client=logs_client,
+        )
+
+    @pytest.mark.aws_validated
+    # TODO LocalStack adds this RequestID to the error response
```

Comment:
Does AWS not have a request ID?

Reply:
No, AWS does not add it, we realised this with the snapshots I believe, we encountered this issue multiple times.

Reply:
As mentioned above, let's take a look once the new provider is in place. :)